File: 프로그래머스/2/148653. 마법의 엘리베이터/마법의 엘리베이터.py
def solution(storey):
    answer = 0
    x = list(map(int, str(storey)[::-1]))

    for i in range(len(x)):
        if x[i] < 5: answer += x[i]
        elif x[i] == 5:
            answer += 5
            if i+1 < len(x) and x[i+1] >= 5: x[i+1] += 1 
        else: 
            answer += 10 - x[i]
            if i+1 < len(x): x[i+1] += 1
            else: answer += 1

    return answer

File: 프로그래머스/2/148653. 마법의 엘리베이터/마법의 엘리베이터.js
function solution(storey) {
    let answer = 0
    const x = String(storey).split("").map(a => Number(a))

    for(let i = x.length-1; i >= 0; i--) {
        if (x[i] > 5) {
            answer += 10 - x[i]
            if (i === 0) answer++
            x[i-1]++
        } else if(x[i] === 5 && i > 0 && x[i-1] >= 5) {
            answer+= 5
            x[i-1]++    
        } else answer += x[i]
    }
    
    return answer
}